packages/medusa/src/services/discount.js

```diff
@@ -182,12 +182,20 @@ class DiscountService extends BaseService {
       const discountRepo = manager.getCustomRepository(this.discountRepository_)
       const ruleRepo = manager.getCustomRepository(this.discountRuleRepository_)
 
+      if (discount.rule?.valid_for) {
```

Comment:
Load products to add them to the set of `valid_for`

Reply:
Can you double check if this is necessary? I think it might suffice to just do this:

```javascript
discount.rule.valid_for = discount.rule.valid_for.map(id => ({ id }))
```